synapsis/src/main/java/com/soen/synapsis/appuser/registration/RegistrationController.java

```diff
@@ -58,6 +60,69 @@ public String register(RegistrationRequest request,
         }
     }
 
+
+    @GetMapping(value ="/admincreation")
+    public String registerAdmin(Model model) {
+        if (isUserAuthenticated()) {
+            model.addAttribute("registrationRequest", new RegistrationRequest());
+
+            return "pages/adminCreationPage";
+        }
+        return null;
+    }
+
+    @PostMapping (value ="/admincreation")
+    public String registerAdmin(RegistrationRequest request,
+                           BindingResult bindingResult,
+                           Model model) {
+
+        try{
+
+            if (bindingResult.hasErrors()) {
+                throw new Exception();
+            }
+
+            if (isUserAuthenticated()){
+                String response = registrationService.registerAdmin(request);
+                return response;
+            }
+        }
+        catch (Exception e) {
+            model.addAttribute("error", "There was an error registering the ADMIN. " + e.getMessage());
+            return registerAdmin(model);
+        }
+        return null;
+    }
+
+
+    @GetMapping("/passwordreset")
+    public String passwordReset(Model model) {
+
```

Comment:
Should we have verification to see if the user is logged in here (as well as in the related post method)?

Reply:
I believe this is more like the password reset done in case someone has forgotten their password?